### PR TITLE
dts: flash: w25qxxdv: Add Device Tree Support for SPI FLASH w25qxxdv

### PR DESCRIPTION
**Pull request by galak**
Convert the w25qxxdv driver to use device tree for SPI device params.
Updated the Arduino 101 config to use device tree to specify the SPI
flash.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by galak** [comment]
@nashif any update on how we name the I2C controllers on arduino_101_sss.  This looks like when we moved away from QMSI I2C to DW I2C.

File: boards/x86/arduino_101/arduino_101.dts [hunk 1]
```diff
@@ -58,6 +58,15 @@
 
 &spi0 {
 	status = "ok";
+	cs-gpios = <&gpio0 24 0>;
```
Comment (nvlsianpu) [inline boards/x86/arduino_101/arduino_101.dts]: I sit possible to have more than one device at the SPI bus? If yes then what with multiple cs declaration and usage?
`#define DT_SPI_FLASH_W25QXXDV_GPIO_SPI_CS_PIN		DT_SNPS_DESIGNWARE_SPI_B0001000_CS_GPIOS_PIN`

**Review by nashif — APPROVED** [review]

File: boards/x86/arduino_101/arduino_101.dts (hunk 1 above)
Comment (galak) [reply boards/x86/arduino_101/arduino_101.dts]: Its possible from a DTS point of view, however we don't handle the code generation side of this correctly today for more than a single device on the spi bus.

Comment (nvlsianpu) [reply boards/x86/arduino_101/arduino_101.dts]: got it

**Pull request merged**